### ISSUE TITLE
Different Embeddings for each run

### ISSUE THREAD

**Issue by DenizEngin1987** [lyst/lightfm#480]
Hello there,

I have constructed a minimal code example presenting my Issue:
`from lightfm import LightFM`
`from scipy.sparse import coo_matrix`

`interactions_dense = [[1.0, 1.0, 1.0, 1.0, 1.0],[1.0, 1.0, 1.0, 1.0, 1.0],[1.0, 1.0, 1.0, 1.0, 1.0],[1.0, 1.0, 1.0, 1.0, 1.0],[1.0, 1.0, 1.0, 1.0, 1.0]]`
`weights_dense = [[5.0, 2.0, 3.0, 4.0, 1.0], [5.0, 4.0, 2.0, 1.0, 2.0],[5.0, 1.0, 1.0, 1.0, 1.0],[5.0, 2.0, 1.0, 1.0, 3.0],[1.0, 1.0, 2.0, 3.0, 1.0]]`

`interactions = coo_matrix (interactions_dense)`
`weights = coo_matrix (weights_dense)`

`model = LightFM(loss='warp', no_components=1)`
`model.fit(interactions, sample_weight=weights, epochs=1)`

`print('user_embeddings: ' + str(model.user_embeddings))`
`print('item_embeddings: ' + `str(model.item_embeddings))`

As one can see, I first create an interaction and a weight matrix and then I train the model with this data. In the end the user-/item-embeddings are printed out.

If I run the code multiple times, I get different embeddings for each run. Is that correct and if so, why is it. 

Thx in advance

**Comment by Med-ELOMARI** [comment]
simply because  a random_state is used when shuffling the data and initializing the parameters like  internal latent representations (embeddings)

https://github.com/lyst/lightfm/blob/4e27c771981d097e3789f536cb2e6a408866546b/lightfm/lightfm.py#L269-L273

as you can see in _initialize() function item and user  features are Initialise by a random state 

**RandomState** _exposes a number of methods for generating random numbers drawn from a variety of probability distributions_ for more info , check [this](https://docs.scipy.org/doc/numpy-1.15.1/reference/generated/numpy.random.RandomState.html)

and moreover you're using  **_epochs=1_** , if you have moooore epochs then embeddings will converge to almost the same value (in theory 🤓 )

Good luck @DenizEngin1987

**Comment by DenizEngin1987** [comment]
Thanks a lot for the help!
